src/meta/processors/job/JobDescription.cpp

```diff
@@ -83,6 +84,11 @@ cpp2::JobDesc JobDescription::toJobDesc() {
 }
 
 bool JobDescription::setStatus(Status newStatus, bool force) {
+  if (status_ == cpp2::JobStatus::FINISHED) {
+    // no one should change the status of a finished job
+    // in case that recovered job running again, but set finished to finished should be ok
+    return newStatus == cpp2::JobStatus::FINISHED;
```

Comment:
1. could add a log here?
2. could wrap the logic to a function:` JobStatus::isUnsettable(status)`  and comment on the function?
```
if ( JobStatus::isUnsettable(status)) {
  return newStatus == status_;
}
```

Reply:
Think it's small enough. better to make put all logic here.

Reply:
I think whether we wrap an function does not depends on how small it is. But about:
1. explicit, make the comment to an function name.
2. consistent, with the `JobStatus::laterThan`. Sentences should have similar abstract level.

Reply:
> I think whether we wrap an function does not depends on how small it is. But about:
> 
> 1. explicit, make the comment to an function name.
> 2. consistent, with the `JobStatus::laterThan`. Sentences should have similar abstract level.

make sence to me.